openhands/runtime/impl/eventstream/eventstream_runtime.py

```diff
@@ -407,6 +406,9 @@ def close(self, close_client: bool = True, rm_all_containers: bool = True):
         if self.session:
             self.session.close()
 
+        if self.attach_to_existing:
```

Comment:
**Suggestion:** Ensure the `docker_client` is closed when `attach_to_existing` is `False`.

**Existing Code:**
```Python
if self.attach_to_existing:
    return
```

**Improved Code:**
```Python
if self.attach_to_existing:
    return
self.docker_client.close()
```

**Details:**
Ensure that the `docker_client` is closed when `attach_to_existing` is `False`.

Reply:
The docker client is actually shared at this point due to use of an LRU cache. (I did this to speed up startup, and tests showed that this did not cause problems).

So closing it is inappropriate at this point.